src/check/ast/node.rs

```diff
@@ -36,9 +36,9 @@ impl From<&Node> for NodeTy {
                 parents: parents.iter().map(ASTTy::from).collect(),
                 body: body.clone().map(ASTTy::from).map(Box::from),
             },
-            Node::Generic { id, isa } => NodeTy::Generic {
+            Node::Generic { id, .. } => NodeTy::Type {
```

Comment:
Not a fan, actually.
`NodeTy` should just copy `Node` so the `AST` isn't unnecessarily changed here.
If we want generate rules these should be in the `generate` stage/module.